scipy/stats/tests/test_distributions.py

```diff
@@ -1203,6 +1214,10 @@ def test_bad_input(self):
         pk = [1, 2, 3]
         assert_raises(ValueError, stats.rv_discrete, **dict(values=(xk, pk)))
 
+        xk = [1, 2, 3, 4, 5]
+        pk = [-0.3, -0.3, 0.3, 0.3, 0.4]
```

Comment:
this test case does not distinguish whether the error is caused by negative values or sum != 1

Reply:
OK. I will do implementation for distinguishable test case (negative value and sum is 1).